### PR TITLE
Quotelinks and Javascript suite

### PR DESCRIPTION
**Pull request by Apogate**
This feels extremely dirty for some reason, I hope it's just because
it's late.

fixes quote linking, for some reason auto-linking wouldn't detect
`&gt;`, but WOULD detect `>` characters. RePod is smirking somewhere in
the distance.

It's an UGLY fix, but it's better than nothing.

More testing tomorrow.

### CODE REVIEW

**Comment by Apogate** [comment]
Quick tests reveal bunch of issues on **/bta/**

I'll re-investigate when i'm awake.

**Comment by Apogate** [comment]
Fixes #145 

Fixes #127

**Comment by Apogate** [comment]
Can't manipulate anything into viable HTML, php or SQL queries. Safe to merge.

**Comment by RePod** [comment]
Might as well clean it up before merging, like the quotes.

**Comment by Apogate** [comment]
Still needs to be fixed:
- [ ] Infinite scroll
- [ ] Entirety of **utility_quotes.js**
- [ ] Everything related to thread updating
- [ ] Quick reply
- [x] Thread stats showing up in index

The first three need heavy work due to their reliance on the old url & uncached page system.

#179

**Comment by RePod** [comment]
> heavy work due to their reliance on the old url & uncached page system.
- Pretty sure infinite scroll just loaded the next page, not hard.
- Utility quotes had nothing to do with URLs and cached pages as they were run on the current page.
- Thread updating worked by loading the literal current URL and looking for new posts based on what it already had.
  - This is why mobile broke it.
- Quick reply probably isn't affected much since it was POSTing to _imgboard.php_.
  - What probably broke is the selector used to trigger the window.
- Thread/Stats. Never 4get ;~;7

**Comment by RePod** [comment]
> +Cookies for name, deletion password and email fields are saved again.

Eventually, if these are being handled by Javascript, these should be stored in Local Storage instead to reduce request overhead.

Since they're being sent twice currently (as part of form data and again as cookies), it's effectively double the information at its best. By switching them to Local Storage, they're only sent once as needed in the original form data.

For example, all of my jQuery suite was recently switched to [storing the configuration options in Local Storage instead of cookies](https://github.com/spootTheLousy/saguaro/blob/master/plugins/jquery/suite_settings.js#L139-L148), thereby not flooding the server with them every request. Contrary to this example, however, the old jQuery files that aren't the menu aren't using Local Storage efficiently but it's better than before.

---

**TL;DR:** Local Storage has a few advantages to Cookies:
- Not sent to the server along with requests.
  - General request size lowered (less sent by client, less received by server)
  - Better privacy (super sekrit client-side only)
- Longer retention.
  - Not necessarily tied to an "expiration" timer like Cookies.
- Supports a few more data types, specifically objects and arrays.
  - Therefore it's easier to store data in a readable and accessible way.
  - For example: `localStorage['postform']['email']` or `localStorage.postform.email`.
- Doesn't need proxy functions (like the current jQuery suite has) to be used easily.
  - However proxy functions are nice and allow a central point of origin.

**Bonus TL;DR:** I'm not stopping you from making your changes, I just know there's a lot beyond your reach that needs to be dealt with. I'll probably do those when you wrap your modifications up.

**Comment by Apogate** [comment]
before and after. minor tweaking, more spacing. bigger font for those with special eyes.

**Comment by Apogate** [comment]
Well infinite scrolling DOES load the next page... It just doesn't append anything to the current document.

**Comment by Apogate** [comment]
Well, I'll wrap this up by saying this push has a lot of important fixes that should probably get merged instead of it stagnating while I work on my jQuery skills.

Merging for now, will revisit the suite down the road.

@RePod I appreciate your patience with me hacking away at your suite.

That's all for now team, tune in next time when I destroy everything else.
